File: akka-samples/akka-sample-remote-scala/src/main/scala/sample/remote/calculator/CalculatorActor.scala
package sample.remote.calculator

import akka.actor.Props
import akka.actor.Actor

class CalculatorActor extends Actor {
  def receive = {
    case Add(n1, n2) =>
      println("Calculating %d + %d".format(n1, n2))
      sender() ! AddResult(n1, n2, n1 + n2)
    case Subtract(n1, n2) =>
      println("Calculating %d - %d".format(n1, n2))
      sender() ! SubtractResult(n1, n2, n1 - n2)
    case Multiply(n1, n2) =>
      println("Calculating %d * %d".format(n1, n2))
      sender() ! MultiplicationResult(n1, n2, n1 * n2)
    case Divide(n1, n2) =>
      println("Calculating %.0f / %d".format(n1, n2))
      sender() ! DivisionResult(n1, n2, n1 / n2)
  }
}

File: akka-samples/akka-sample-remote-java/src/main/java/sample/remote/calculator/CalculatorActor.java
package sample.remote.calculator;

import akka.actor.UntypedActor;

public class CalculatorActor extends UntypedActor {
  @Override
  public void onReceive(Object message) {

    if (message instanceof Op.Add) {
      Op.Add add = (Op.Add) message;
      System.out.println("Calculating " + add.getN1() + " + " + add.getN2());
      Op.AddResult result = new Op.AddResult(add.getN1(), add.getN2(),
          add.getN1() + add.getN2());
      getSender().tell(result, getSelf());

    } else if (message instanceof Op.Subtract) {
      Op.Subtract subtract = (Op.Subtract) message;
      System.out.println("Calculating " + subtract.getN1() + " - "
          + subtract.getN2());
      Op.SubtractResult result = new Op.SubtractResult(subtract.getN1(),
          subtract.getN2(), subtract.getN1() - subtract.getN2());
      getSender().tell(result, getSelf());

    } else if (message instanceof Op.Multiply) {
      Op.Multiply multiply = (Op.Multiply) message;
      System.out.println("Calculating " + multiply.getN1() + " * "
          + multiply.getN2());
      Op.MultiplicationResult result = new Op.MultiplicationResult(
          multiply.getN1(), multiply.getN2(), multiply.getN1()
              * multiply.getN2());
      getSender().tell(result, getSelf());

    } else if (message instanceof Op.Divide) {
      Op.Divide divide = (Op.Divide) message;
      System.out.println("Calculating " + divide.getN1() + " / "
          + divide.getN2());
      Op.DivisionResult result = new Op.DivisionResult(divide.getN1(),
          divide.getN2(), divide.getN1() / divide.getN2());
      getSender().tell(result, getSelf());

    } else {
      unhandled(message);
    }
  }
}
